crates/protocol/src/registry.rs

```diff
@@ -76,6 +77,93 @@ impl VisitBytes for LogLeaf {
     }
 }
 
+/// Represents a valid package identifier in the registry.
+///
+/// Valid package identifiers conform to the component model specification
+/// for identifiers.
+///
+/// A valid component model identifier is the format `<namespace>:<name>`,
+/// where both parts are also valid WIT identifiers (i.e. kebab-cased).
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Ord, PartialOrd)]
+pub struct PackageId {
+    id: String,
+    colon: usize,
+}
+
+impl PackageId {
+    /// Creates a package identifier from the given string.
+    ///
+    /// Returns an error if the given string is not a valid package identifier.
+    pub fn new(id: impl Into<String>) -> anyhow::Result<Self> {
```

Comment:
Should this be called `try_new` and/or does it even need to exist outside the parse & serde impls?

Reply:
This exists for anyone that wants to move the string into the `PackageId`; in low-level libraries like this, I generally like to give people the option of not cloning if possible.

Happy to call it `try_new` (I feel like the naming around fallible a `new` isn't consistent in a lot of codebases, though; `wasmtime` has a bunch of `new` that return `T` and also `Result<T>`).

Reply:
Yeah fallible `new` is quite common.